src/main/java/org/junit/runners/parameterized/BlockJUnit4ClassRunnerWithParameters.java

```diff
@@ -73,6 +87,30 @@ private Object createTestUsingFieldInjection() throws Exception {
         return testClassInstance;
     }
 
+    private boolean constructorHasParameters() {
+        Class<?>[] parameterTypes = getTestClass().getOnlyConstructor().getParameterTypes();
+        return parameterTypes.length > 0;
+    }
+    
+    @Override
+    protected Statement methodInvoker(FrameworkMethod method,
+            Object testInstance) {
+        if(useTestInjection) {
+            return new InvokeParameterizedMethod(method, parameters, testInstance);
+        } else {
+            return new InvokeMethod(method, testInstance);
+        }
+    }
+    
+    @Override
+    protected void validateTestMethods(List<Throwable> errors) {
+        List<FrameworkMethod> methods = getTestClass().getAnnotatedMethods(Test.class);
+
+        for (FrameworkMethod eachTestMethod : methods) {
+            eachTestMethod.validatePublicVoid(false, errors);
+        }
```

Comment:
This should delegate to super by default, but additionally check whether the test methods have the correct parameters in case `useTestInjection` is `true`.